### PR TITLE
[PW-2689] 3DS1 & 3DS2 payments - Handle the backend

### PR DESCRIPTION
**Pull request by cyattilakiss**
## Summary
Handle the endpoint accepting the state.data and do the paymentDetails request

### CODE REVIEW

File: src/Handlers/PaymentResponseHandler.php [hunk 1]
```diff
@@ -57,101 +69,208 @@ class PaymentResponseHandler
...
+    public function handleShopwareAPIs(
+        AsyncPaymentTransactionStruct $transaction,
+        SalesChannelContext $salesChannelContext,
+        PaymentResponseHandlerResult $paymentResponseHandlerResult
+    ) {
+        $orderTransactionId = $transaction->getOrderTransaction()->getId();
+        $resultCode = $paymentResponseHandlerResult->getResultCode();
+        $context = $salesChannelContext->getContext();
+
+        // Only store psp reference for the transaction if this is the firs/original pspreference
```
Comment (acampos1916) [inline src/Handlers/PaymentResponseHandler.php:193]:
```suggestion
        // Only store psp reference for the transaction if this is the first/original pspreference
```

File: src/Handlers/PaymentResponseHandler.php [hunk 2]
```diff
@@ -67,120 +88,197 @@ public function __construct(
...
                 // Unsupported resultCode
                 $this->logger->error(
-                    "There was an error with the payment method. id:  " . $orderTransactionId .
+                    "There was an error with the payment method. id:  " .
                     ' Unsupported result code in response: ' . print_r($response, true)
                 );
+        }
+    }
+
+    public function handleShopwareAPIs(
```
Comment (acampos1916) [inline src/Handlers/PaymentResponseHandler.php:175]: Just for consistency with the second function
```suggestion
    public function handleShopwareApis(
```

File: src/Handlers/CardsPaymentMethodHandler.php [hunk 3]
```diff
@@ -203,7 +203,11 @@ public function pay(
             );
         }
 
-        return $this->paymentResponseHandler->handlePaymentResponse($response, $transaction, $salesChannelContext);
+        $orderNumber = $transaction->getOrder()->getOrderNumber();
+
+        $result = $this->paymentResponseHandler->handlePaymentResponse($response, $orderNumber, $salesChannelContext);
+
+        return $this->paymentResponseHandler->handleShopwareAPIs($transaction, $salesChannelContext, $result);
```
Comment (acampos1916) [inline src/Handlers/CardsPaymentMethodHandler.php:210]: Just for consistency with the second function
```suggestion
        return $this->paymentResponseHandler->handleShopwareApis($transaction, $salesChannelContext, $result);
```

**Comment by cyattilakiss** [comment]
@msilvagarcia we are going to merge this one for now because @acampos1916 and I also depend on this PR for our current tickets. Could you please review it even after it's merged and flag anything you think should be changed? I can do that in a separate PR then.
I hope this can work for you as well :)

**Pull request merged**